traits/tests/test_trait_notifiers.py

```diff
@@ -0,0 +1,38 @@
+# (C) Copyright 2005-2023 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+from traits.api import HasTraits, Int
+
+
+class TestTraitNotifiers(unittest.TestCase):
+    def test_ui_dispatch(self):
+        # Given
+        class DispatchTest(HasTraits):
+            test_param = Int()
+        t = DispatchTest()
+
+        event_list = []
+
+        # create handler
+        def test_handler(event):
+            event_list.append(event)
+
+        # When
+        t.observe(test_handler, 'test_param', dispatch='ui')
+
+        # Then
+        try:
+            t.test_param = 1
+        except Exception:
```

Comment:
Let's remove this try/except - it's just extra boilerplate and we don't really need it. If an exception is raised, the test will already fail, and there's no particular reason to expect an exception to be raised either before or after the change in this PR.

Reply:
I'd also move the `t.test_param = 1` to the "When" block - it's the main part of the behaviour we're testing. The "Then" block is for testing the effects of that behaviour.

Reply:
It's often helpful to think about expanding out the "When/Then" to full sentence fragments: E.g., "**When** we change the value of a trait with a `dispatch='ui'` observer, *then* that observer is run as expected.

Reply:
Thanks! thats a great advice